### ISSUE TITLE
Error with robustroi.nii.gz - PreFreeSurferPipeline

### ISSUE THREAD

**Issue by alexsayal** [Washington-University/HCPpipelines#154]
Hello all,

I am getting this error while starting PreFreeSurferPipeline on the Example dataset:

```
Image Exception : #22 :: Failed to read volume /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/robustroi.nii.gz
Error : Error: short read, file may be truncated
Failed to read volume /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/robustroi.nii.gz
Error : Error: short read, file may be truncated
Could not open matrix file /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/roi2std.mat
Cannot read concat-matrix
Mon Oct 28 11:26:50 WET 2019:ACPCAlignment.sh: While running '/home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts/ACPCAlignment.sh --workingdir=/home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment --in=/home/rabreu/Downloads/ExampleData/100307/T1w/T1w --ref=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm.nii.gz --out=/home/rabreu/Downloads/ExampleData/100307/T1w/T1w_acpc --omat=/home/rabreu/Downloads/ExampleData/100307/T1w/xfms/acpc.mat --brainsize=150':
Mon Oct 28 11:26:50 WET 2019:ACPCAlignment.sh: ERROR: '/usr/local/fsl/bin/convert_xfm' command failed with return code: 253
Mon Oct 28 11:26:50 WET 2019:PreFreeSurferPipeline.sh: While running '/home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/PreFreeSurferPipeline.sh --path=/home/rabreu/Downloads/ExampleData --subject=100307 --t1=/home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T1w_MPR1/100307_3T_T1w_MPR1.nii.gz@ --t2=/home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T2w_SPC1/100307_3T_T2w_SPC1.nii.gz@ --t1template=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm.nii.gz --t1templatebrain=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm_brain.nii.gz --t1template2mm=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_2mm.nii.gz --t2template=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_0.7mm.nii.gz --t2templatebrain=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_0.7mm_brain.nii.gz --t2template2mm=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_2mm.nii.gz --templatemask=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm_brain_mask.nii.gz --template2mmmask=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_2mm_brain_mask_dil.nii.gz --brainsize=150 --fnirtconfig=/home/rabreu/Downloads/HCPpipelines/global/config/T1_2_MNI152_2mm.cnf --fmapmag=/home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T1w_MPR1/100307_3T_FieldMap_Magnitude.nii.gz --fmapphase=/home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T1w_MPR1/100307_3T_FieldMap_Phase.nii.gz --fmapgeneralelectric=NONE --echodiff=2.46 --SEPhaseNeg=NONE --SEPhasePos=NONE --seechospacing=NONE --seunwarpdir=NONE --t1samplespacing=0.0000074 --t2samplespacing=0.0000021 --unwarpdir=z --gdcoeffs=NONE --avgrdcmethod=NONE --topupconfig=NONE --printcom=':
Mon Oct 28 11:26:50 WET 2019:PreFreeSurferPipeline.sh: ERROR: '/home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts/ACPCAlignment.sh' command failed with return code: 253
```

I am testing this on a Ubuntu 18.04 machine with FSL 6.0.2 and FreeSurfer 6 installed. I have tried some workarounds without success.

Could you please help me find the error?

Thank you in advance!

**Comment by glasserm** [comment]
How did you call the PreFreeSurfer pipeline and how did you set up the HCP Pipelines setup script?

**Comment by alexsayal** [comment]
I am calling PreFreeSurfer with `./PreFreeSurferPipelineBatch.sh` after editing the file with the correct folders for my setup:

```
StudyFolder="${HOME}/Downloads/ExampleData"
Subjlist="100307"
EnvironmentScript="${HOME}/Downloads/HCPpipelines/Examples/Scripts/SetUpHCPPipeline.sh"
```
I also set `AvgrdcSTRING="NONE"`.

In the SetUpHCPPipeline script I edited the variables HCPPIPEDIR and CARET7DIR to match the correct folders. FSL is already in the PATH.
```
export HCPPIPEDIR="${HOME}/Downloads/HCPpipelines"
export CARET7DIR="${HOME}/SOFTWARE/workbench/bin_linux64"
```

**Comment by glasserm** [comment]
I would paste in those files in their entirety.

**Comment by alexsayal** [comment]
Here are the files (in .txt for sending)
[PreFreeSurferPipelineBatch.txt](https://github.com/Washington-University/HCPpipelines/files/3779288/PreFreeSurferPipelineBatch.txt)
[SetUpHCPPipeline.txt](https://github.com/Washington-University/HCPpipelines/files/3779349/SetUpHCPPipeline.txt)

**Comment by glasserm** [comment]
What do the full logfile show?

**Comment by alexsayal** [comment]
This are the contents of file PreFreeSurferPipeline.sh.o16003:
```
PreFreeSurferPipeline.sh: HCPPIPEDIR: /home/rabreu/Downloads/HCPpipelines
PreFreeSurferPipeline.sh: FSLDIR: /usr/local/fsl
PreFreeSurferPipeline.sh: HCPPIPEDIR_Global: /home/rabreu/Downloads/HCPpipelines/global/scripts
PreFreeSurferPipeline.sh: HCPPIPEDIR_PreFS: /home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts
Mon Oct 28 17:00:31 WET 2019:PreFreeSurferPipeline.sh: Platform Information aFollows: 
Linux CELSIUS-W530 5.0.0-32-generic #34~18.04.2-Ubuntu SMP Thu Oct 10 10:36:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
Mon Oct 28 17:00:31 WET 2019:PreFreeSurferPipeline.sh: Parsing Command Line Options
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Finished Parsing Command Line Options
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: StudyFolder: /home/rabreu/Downloads/ExampleData
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Subject: 100307
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wInputImages: /home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T1w_MPR1/100307_3T_T1w_MPR1.nii.gz@
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wInputImages: /home/rabreu/Downloads/ExampleData/100307/unprocessed/3T/T2w_SPC1/100307_3T_T2w_SPC1.nii.gz@
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wTemplate: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wTemplateBrain: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm_brain.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wTemplate2mm: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_2mm.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wTemplate: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_0.7mm.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wTemplateBrain: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_0.7mm_brain.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wTemplate2mm: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T2_2mm.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: TemplateMask: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm_brain_mask.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Template2mmMask: /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_2mm_brain_mask_dil.nii.gz
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: BrainSize: 150
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: FNIRTConfig: /home/rabreu/Downloads/HCPpipelines/global/config/T1_2_MNI152_2mm.cnf
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: MagnitudeInputName: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: PhaseInputName: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: GEB0InputName: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: TE: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: SpinEchoPhaseEncodeNegative: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: SpinEchoPhaseEncodePositive: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: SEEchoSpacing: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: SEUnwarpDir: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wSampleSpacing: 0.0000074
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wSampleSpacing: 0.0000021
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: UnwarpDir: z
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: GradientDistortionCoeffs: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: AvgrdcSTRING: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: TopupConfig: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: BiasFieldSmoothingSigma: 
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: UseJacobian: true
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wBiasCorrect: 
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: CustomBrain: NONE
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: ProcessingMode: HCPStyleData
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: FSLDIR: /usr/local/fsl
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: HCPPIPEDIR: /home/rabreu/Downloads/HCPpipelines
========================================
  DIRECTORY: /home/rabreu/Downloads/HCPpipelines
    PRODUCT: HCP Pipeline Scripts
    VERSION: Post-v4.0.1
========================================
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: HCPPIPEDIR_Global: /home/rabreu/Downloads/HCPpipelines/global/scripts
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: HCPPIPEDIR_PreFS: /home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T1wFolder: /home/rabreu/Downloads/ExampleData/100307/T1w
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: T2wFolder: /home/rabreu/Downloads/ExampleData/100307/T2w
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: AtlasSpaceFolder: /home/rabreu/Downloads/ExampleData/100307/MNINonLinear
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: POSIXLY_CORRECT=
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Processing Modality: T1w
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: NOT PERFORMING GRADIENT DISTORTION CORRECTION
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Not Averaging T1w Images
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: ONLY ONE IMAGE FOUND: COPYING
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: Aligning T1w image to /home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm.nii.gz to create native volume space
Mon Oct 28 17:00:33 WET 2019:PreFreeSurferPipeline.sh: mkdir -p /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment
ACPCAlignment.sh: HCPPIPEDIR: /home/rabreu/Downloads/HCPpipelines
ACPCAlignment.sh: FSLDIR: /usr/local/fsl
Mon Oct 28 17:00:33 WET 2019:ACPCAlignment.sh: START
Final FOV is: 
0.000000 256.000000 0.000000 320.000000 58.000000 214.000000 

Mon Oct 28 17:00:45 WET 2019:ACPCAlignment.sh: ERROR: '/usr/local/fsl/bin/convert_xfm' command failed with return code: 253

===> ERROR: Command returned with nonzero exit code
---------------------------------------------------
         script: ACPCAlignment.sh
stopped at line: 109
           call: ${FSLDIR}/bin/convert_xfm -omat "$WD"/full2std.mat -concat "$WD"/roi2std.mat "$WD"/full2roi.mat
  expanded call: /usr/local/fsl/bin/convert_xfm -omat /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/full2std.mat -concat /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/roi2std.mat /home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment/full2roi.mat
      exit code: 253
---------------------------------------------------

===> Aborting execution!

Mon Oct 28 17:00:45 WET 2019:PreFreeSurferPipeline.sh: ERROR: '/home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts/ACPCAlignment.sh' command failed with return code: 253

===> ERROR: Command returned with nonzero exit code
---------------------------------------------------
         script: PreFreeSurferPipeline.sh
stopped at line: 602
           call: ${RUN} ${HCPPIPEDIR_PreFS}/ACPCAlignment.sh --workingdir=${TXwFolder}/ACPCAlignment --in=${TXwFolder}/${TXwImage} --ref=${TXwTemplate} --out=${TXwFolder}/${TXwImage}_acpc --omat=${TXwFolder}/xfms/acpc.mat --brainsize=${BrainSize}
  expanded call: /home/rabreu/Downloads/HCPpipelines/PreFreeSurfer/scripts/ACPCAlignment.sh --workingdir=/home/rabreu/Downloads/ExampleData/100307/T1w/ACPCAlignment --in=/home/rabreu/Downloads/ExampleData/100307/T1w/T1w --ref=/home/rabreu/Downloads/HCPpipelines/global/templates/MNI152_T1_0.7mm.nii.gz --out=/home/rabreu/Downloads/ExampleData/100307/T1w/T1w_acpc --omat=/home/rabreu/Downloads/ExampleData/100307/T1w/xfms/acpc.mat --brainsize=150
      exit code: 253
---------------------------------------------------

===> Aborting execution!

```

**Comment by glasserm** [comment]
Can you run any of those FSL binaries outside of the HCP Pipelines?  Code 253 seems to be a generic "cannot read configuration file" error code.

**Comment by alexsayal** [comment]
Well convert_xfm is available and I can run it. I do not have the matrix `roi2std.mat`, but the following line works without the --concat option:
```
${FSLDIR}/bin/convert_xfm -omat ${WD}/full2std.mat ${WD}full2roi.mat 
```

**Comment by mharms** [comment]
We’ve been making a lot of changes lately, and I wonder if your particular copy of the master didn’t catch a stage with a bug.

Try using the very latest master if you want to operate at the very tip of the repo, or even better, consider waiting a week until we have things stabilized.

Cheers,
-MH

--
Michael Harms, Ph.D.
-----------------------------------------------------------
Associate Professor of Psychiatry
Washington University School of Medicine
Department of Psychiatry, Box 8134
660 South Euclid Ave.                        Tel: 314-747-6173
St. Louis, MO  63110                          Email: mharms@wustl.edu

**Comment by coalsont** [comment]
You could also try a release version from before we started merging things,
such as v4.0.1.

Tim

**Comment by alexsayal** [comment]
I am using the latest master version. I also tried Release 4.0.1 but the same error occurs.

Thank you all for the suggestions, I will wait for all your changes!

**Comment by glasserm** [comment]
Is there anything in stderr?

**Comment by mharms** [comment]
Well, if v4.0.1 isn’t working, that suggests a problem with your installation, not a bug in the code

--
Michael Harms, Ph.D.
-----------------------------------------------------------
Associate Professor of Psychiatry
Washington University School of Medicine
Department of Psychiatry, Box 8134
660 South Euclid Ave.                        Tel: 314-747-6173
St. Louis, MO  63110                          Email: mharms@wustl.edu

**Comment by alexsayal** [comment]
Hello all,
I reinstalled everything (again) and the bug is solved - running v4.0.1.
Thank you for your help on this!

**Comment by glasserm** [comment]
Sounds good.
